text/0028-ed25519-validation/0028-ed25519-validation.md

```diff
@@ -0,0 +1,79 @@
+
+In contrast to the Section [Decoding](https://tools.ietf.org/html/rfc8032#section-5.1.3) of RFC 8032, it is _not_ required that the encodings of A and R are canonical. As long as the corresponding (x,y) is a valid curve point, any of such edge cases will be accepted.
+
+It is worth noting that due to allowing different encodings of the same point, one cannot check point equality by doing byte per byte comparisons.
+
+## Validation 
+
+The RFC 8032 mentions two alternative verification equations:
+1. [8][S]B = [8]R + [8][k]A'
+2. [S]B = R + [k]A'
+
+Each honestly generated signature following RFC 8032 satisfies the second, cofactor-less equation and thus, also the first equation. However, the opposite is not true: There are solutions only satisfying the first but not the latter.<br> This ambiguity in RFC 8032 has led to the current situation in which different implementations rely on different verification equations. 
+
+In order to be consistent with batched verification, the group equation [8][S]B = [8]R + [8][k]A' _must_ be used for validations instead of [S]B = R + [k]A'.
+
+## Malleability
+
+The non-negative integer S is encoded into 32 bytes as part of the signature. However, a third party could replace S with S' = S + n·L for any natural n with S' < 2<sup>256</sup> and the modified signature R || S' would still pass verification. Requiring a value less than L resolves this malleability issue. Unfortunately, this check is not present in all common Ed25519 implementations.
+
+Analogous to RFC 8032, the encoding of S _must_ represent an integer less than L.
+
+It is not possible for an external party to mutate R and still pass verification. The owner of the secret key, however, can create many different signatures for the same content: While Ed25519 defines a deterministic method of calculating the integer scalar r from the private key and the message, it is impossible to tell during signature verification if the point R = [r]B was created properly or any other scalar has been used.<br> As a result, there is a practically countless amount of different valid signatures corresponding to a certain message and public key.
+
+# Drawbacks
+
+- Allowing non-canonical encodings is a direct contradiction of RFC 8032 and rather unintuitive. Furthermore, it introduces alternative encodings for a handful of points on the curve. Even though such points will, for all practical purposes, never occur in honest signatures, it still theoretically introduces an external party malleability vector.
+- The cofactor validation is computationally slightly more expensive than the cofactor-less version since it requires a multiplication by 8.
+
+# Rationale and alternatives
+
+While the malleability of S poses serious issues and thus, must be prevented, the other two validation criteria, namely non-canonical encodings as well as the cofactor-less validation equation, could be relaxed without introducing attack vectors.
```

Comment:
How about: "the other two validation criteria can be relaxed, namely non-canonical encodings as well as the cofactor-less validation equation, can be used since they do not cause the same issues"

Reply:
Relaxed in which way? You mean:

1. Accept only canonical encodings for A and R
3. Use verification condition O ?= -[s]B + R + [k]A

This might render old and accepted signatures as invalid.

Reply:
Made it clear that even when allowing non-canonical encodings of A and R do not lead to malleability by a public attacker as a malleable S would.

Reply:
Great, thanks